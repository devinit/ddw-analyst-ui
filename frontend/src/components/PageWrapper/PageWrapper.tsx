import * as React from 'react';
import classNames from 'classnames';

interface PageWrapperProps {
  fullPage?: boolean;
}

export const PageWrapper: React.SFC<PageWrapperProps> = props => {
  return (
    <div className={ classNames('wrapper', { 'wrapper-full-page': props.fullPage }) } >
      { props.children }
    </div>
  );
};
