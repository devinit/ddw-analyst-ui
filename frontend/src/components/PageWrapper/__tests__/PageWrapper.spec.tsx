import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { PageWrapper } from '../PageWrapper';

test('renders with no children', () => {
  const renderer = TestRenderer.create(<PageWrapper />).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders with children', () => {
  const renderer = TestRenderer.create(<PageWrapper>My Child</PageWrapper>).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders with the wrapper-full-page class when set to fullPage', () => {
  const renderer = TestRenderer.create(<PageWrapper fullPage>My Child</PageWrapper>).toJSON();

  expect(renderer).toMatchSnapshot();
});
