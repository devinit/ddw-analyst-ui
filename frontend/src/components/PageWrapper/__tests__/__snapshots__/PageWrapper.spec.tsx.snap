// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with children 1`] = `
<div
  className="wrapper"
>
  My Child
</div>
`;

exports[`renders with no children 1`] = `
<div
  className="wrapper"
/>
`;

exports[`renders with the wrapper-full-page class when set to fullPage 1`] = `
<div
  className="wrapper wrapper-full-page"
>
  My Child
</div>
`;
